parquet-generator/src/main/resources/ByteBitPacking512VectorLE

```diff
@@ -0,0 +1,3095 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.parquet.column.values.bitpacking;
+
+import jdk.incubator.vector.ByteVector;
+import jdk.incubator.vector.IntVector;
+import jdk.incubator.vector.LongVector;
+import jdk.incubator.vector.ShortVector;
+import jdk.incubator.vector.Vector;
+import jdk.incubator.vector.VectorMask;
+import jdk.incubator.vector.VectorOperators;
+import jdk.incubator.vector.VectorShuffle;
+import jdk.incubator.vector.VectorSpecies;
+
+import java.nio.ByteBuffer;
+
+/**
+ * This is an auto-generated source file and should not edit it directly.
+ */
+public abstract class ByteBitPacking512VectorLE {
```

Comment:
Should we remove this generated file from git? I assume it will be automatically created at maven build time, like other generators do.

Reply:
This is a good question! 
    Why I put the generated file in the resources directory? Due to use Java Vector, it needs rearrange/shuffle/lanewise operations, so it is difficult to create automatically ByteBitPacking512VectorLE like other generators do. On the contrary,  It is relatively simple to code directly instead of generating.
    I think more important is the finished code than how that code was generated. 
    In fact, I've done part of the work to generate ByteBitPacking512VectorLE like other generators do. But is the generating valuable if it if it increases the workload ?
@wgtmac Do you think it is necessary to generate ByteBitPacking512VectorLE like other generators do ?  thanks.

Reply:
OK, I got your point now. Thanks!

Reply:
Do you have any script to generate the code here? If true, it would be great to commit it as well.

Reply:
@wgtmac I have the script, but it generate only the code partly. 
It needs hard work and lots of time to complete the script(I don't think it is necessary). 
In fact, the code is completed mostly by manually.
Should I commit the script which is partly completed ?

Reply:
In this case, the script is not necessary. Manual bit-unpacking code is error-prone, we really rely on the quality and coverage of test cases.

Reply:
> In this case, the script is not necessary. Manual bit-unpacking code is error-prone, we really rely on the quality and coverage of test cases.

@wgtmac  I strongly agree with you, so I try my best to cover all aspects(different aspects from bitWidth 1 to 32) in class TestByteBitPacking512VectorLE. Besides, I have done the TPC-H testing, and compared the query result with before optimization. In short, I have done more work to ensure the code quality.